### PR TITLE
Grouping option on Typeaheads

### PR DESCRIPTION
**Pull request by Michele-Masciave**
Added possibility to group results in the Typeaheads.

### CODE REVIEW

**Review by neotob — DISMISSED** [review]

File: src/lib/helpers/typeahead.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  if (isStringArray) {
+    return options as string[];
+  }
+
+  return (options as Record<string, string>[]).map((option) => option.value);
+};
+
+const renderMenu = (results: LabelValueOption[], menuProps: RenderMenuProps): JSX.Element => {
+  const anonymousOptions: LabelValueOption[] = [];
```
Comment (manni497) [inline src/lib/helpers/typeahead.tsx:18]: This code can be written way better.. can you try to use the new Set(groups) to get distinct groups and directly render them?

**Review by manni497 — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -44,7 +44,8 @@
     "start": "rollup -c -w",
     "start-all": "concurrently \"yarn start\" \"yarn start-yalc\"",
     "start-yalc": "yarn nodemon --watch dist -x \"yarn yalc push\"",
-    "storybook": "start-storybook -p 6006"
+    "storybook": "start-storybook -p 6006",
+    "test": "cd cypress && yarn cyr"
```
[the remark this answers is not in the document]
Comment (manni497) [reply package.json:48]: Ahh now I got what you mean, there is a yarn test in the prepare-pr without the command.. Please remove this line, I will check with Timothy and properly fix this

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ## [Unreleased]
 
+### Added
+
+- The property `group` in the `LabelValueOption`
+- The possibility to group results both in the `AsyncTypeaheadInput` and `StaticTypeaheadInput` using the property `useGroupBy`
+
+### Changed
+
+- The `helpers/typeahead` format from `ts` to `tsx`
```
[the remark this answers is not in the document]
Comment (Michele-Masciave) [reply CHANGELOG.md]: Done

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (Michele-Masciave) [reply package.json]: exactly, done

File: src/lib/helpers/typeahead.tsx (hunk 1 above)
Comment (Michele-Masciave) [reply src/lib/helpers/typeahead.tsx]: done

**Review by manni497 — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Corrected the cypress test running script
```
[the remark this answers is not in the document]
Comment (Michele-Masciave) [reply CHANGELOG.md]: done

**Review by manni497 — APPROVED** [review]

**Pull request merged**